slugger.cabal

```diff
@@ -0,0 +1,63 @@
+cabal-version:        3.0
+name:                 slugger
+version:              0.1.0.0
+synopsis:             Clean URI slugs for Haskell
+description:          Clean URI slugs for Haskell
+homepage:             https://github.com/rpearce/slugger
+bug-reports:          https://github.com/rpearce/slugger/issues
+license:              BSD-3-Clause
+license-file:         LICENSE
+author:               Robert W. Pearce
+maintainer:           me@robertwpearce.com
+copyright:            2021 Robert W. Pearce
+category:             Data, Text, Web
+extra-source-files:   CHANGELOG.md
+                    , README.md
+
+common lang
+    default-language: Haskell2010
+
+common deps
+    build-depends:
+        base ^>= 4.13
+      , text ^>= 1.2
+      , text-icu ^>= 0.7
```

Comment:
I think the version bounds are overly restrictive, I don't think this library can be built anything other than GHC 8.8. There are two camps about upper bounds in Haskell ecosystem.

* Either you put a lower bound for the oldest supported version, and an upper bound for the newest version you *tested*
* Or you put a lower bound for the oldest supported version, and only put an upper bound when you *know* that the library doesn't work with newer versions.

I prefer the second one, because the first one tend to cause unnecessary work when libraries get updated. So I'd change this to something like:

```
        base >= 4.13
      , text >= 1.2
      , text-icu >= 0.7
```

Reply:
The hackage people insisted, as part of giving me the ability to contribute to hackage, that I add an upper bound to dependencies.

I doubt there is, but is there a way to test older versions of deps quickly? Or do I need to go back to previous versions and try them out and see how far back the library will still work?

Reply:
Oh, okay. But still if you're going to put an upper bound, at least make it a bit wide.

> as part of giving me the ability to contribute to hackage, that I add an upper bound to dependencies.

Them's fighting words :). I'll complain about this policy somewhere. I spend hours an hours after each GHC update to send PR's to many libraries just to relax their upper bound in order to get them compile on GHC 9. I don't see how the upper bounds are helping, especially without tooling to have them up to date.

> I doubt there is, but is there a way to test older versions of deps quickly?

I don't know. Once you upload it to Hackage, there is [a Matrix builder](https://matrix.hackage.haskell.org/#/package/accelerate) that tests on different compilers, but not with different versions of dependencies. As I said, I tend to just put a lower bound on `base`, and only use lower and upper bounds once I learn that it doesn't work with a certain version.

>  Or do I need to go back to previous versions and try them out and see how far back the library will still work

My opinion is that I wouldn't worry too much about supporting more than the last 3 GHC versions. You are already testing on `GHC 8.8.4`, and that is enough of a lower bound. So I would remove the overrides from `flake.nix`, see if it compiles (without the bounds), and use the major versions of the libraries there as the lower bound. And I'd compile it with the latest GHC (`GHC 9.0.2`), and use the package versions there, plus one major version as the upper bound.

Reply:
Okay, I'll give this a go in a few hours! You are awesome for guiding me through this